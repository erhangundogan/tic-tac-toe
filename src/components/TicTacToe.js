import React from 'react';
import GameBoard from './GameBoard';
import useTicTacToe from '../hooks/useTicTacToe';
import InfoBoard from './InfoBoard';
import './styles.css';

const TicTacToe = () => {
  const {
    winsCount,
    drawsCount,
    restart,
    isFinished,
    isPlayersTurn,
    board,
    onClick
  } = useTicTacToe();

  return (
    <div className="container">
      <GameBoard { ...{ board, onClick } } />
      <InfoBoard { ...{
        winsCount,
        drawsCount,
        restart,
        isFinished,
        isPlayersTurn } }
      />
    </div>
  );
};

export default TicTacToe;
