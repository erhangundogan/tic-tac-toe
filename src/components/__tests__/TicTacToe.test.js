import React from 'react';
import { render } from '@testing-library/react';
import TicTacToe from '../TicTacToe';

describe('TicTacToe', () => {
  it('renders', () => {
    const { container } = render(<TicTacToe />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
