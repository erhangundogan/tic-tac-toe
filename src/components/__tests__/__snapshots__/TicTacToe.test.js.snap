// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TicTacToe renders 1`] = `
<div
  class="container"
>
  <div>
    <div
      class="col-numbers"
    >
      <div>
        0
      </div>
      <div>
        1
      </div>
      <div>
        2
      </div>
    </div>
    <div
      class="row-numbers-and-board"
    >
      <div
        class="row-numbers"
      >
        <div>
          0
        </div>
        <div>
          1
        </div>
        <div>
          2
        </div>
      </div>
      <table
        class="board"
      >
        <tbody
          id="board"
        >
          <tr>
            <td
              class="cell"
              id="00"
            />
            <td
              class="cell"
              id="01"
            />
            <td
              class="cell"
              id="02"
            />
          </tr>
          <tr>
            <td
              class="cell"
              id="10"
            />
            <td
              class="cell"
              id="11"
            />
            <td
              class="cell"
              id="12"
            />
          </tr>
          <tr>
            <td
              class="cell"
              id="20"
            />
            <td
              class="cell"
              id="21"
            />
            <td
              class="cell"
              id="22"
            />
          </tr>
        </tbody>
      </table>
    </div>
  </div>
  <div
    class="info"
  >
    <div>
      X's turn
    </div>
    <div>
      X has no winning!
    </div>
    <div>
      O has no winning!
    </div>
    <div />
    <div />
  </div>
</div>
`;
